common/http/platform/beast/http.cpp

```diff
@@ -177,16 +188,48 @@ void Client::ResolveHandler(
 
 	weak_ptr<Client> weak_client(stream_active_);
 
-	this->stream_.async_connect(
-		resolver_results_,
-		[weak_client](const error_code &err, const asio::ip::tcp::endpoint &endpoint) {
-			auto client = weak_client.lock();
-			if (client) {
-				client->ConnectHandler(err, endpoint);
-			}
-		});
+	beast::get_lowest_layer(this->stream_)
+		.async_connect(
+			resolver_results_,
+			[this, weak_client](const error_code &err, const asio::ip::tcp::endpoint &endpoint) {
+				auto client = weak_client.lock();
+				if (client) {
+					if (is_https_) {
+						return client->HandshakeHandler(err, endpoint);
+					}
+					return client->ConnectHandler(err, endpoint);
+				}
+			});
+}
+
+void Client::HandshakeHandler(const error_code &err, const asio::ip::tcp::endpoint &endpoint) {
+	if (err) {
+		CallErrorHandler(err, request_, header_handler_);
+		return;
+	}
+
+	// Set SNI Hostname (many hosts need this to handshake successfully)
+	if (!SSL_set_tlsext_host_name(stream_.native_handle(), request_->address_.host.c_str())) {
+		beast::error_code ec {static_cast<int>(::ERR_get_error()), asio::error::get_ssl_category()};
+		logger_.Error("Failed to set SNI host name: " + ec.message());
+	}
+
+	stream_.async_handshake(ssl::stream_base::client, [this, err, endpoint](const error_code &ec) {
+		if (ec) {
+			logger_.Error("https: Failed to perform the SSL handshake: " + ec.message());
+			CallErrorHandler(err, request_, header_handler_);
+			return;
+		}
+		weak_ptr<Client> weak_client(stream_active_);
+		logger_.Debug("https: Successful SSL handshake");
+		auto client = weak_client.lock();
+		if (client) {
+			client->ConnectHandler(err, endpoint);
```

Comment:
`ec` here also. Sorry, the naming is confusing, apparently I called it `err` inside `ConnectHandler`, but it *is* `ec`.